lib/components/steps/mapping/MappingSearch.tsx

```diff
@@ -7,33 +7,38 @@ import { useDebounce } from "../../../hooks.ts";
 
 interface MappingSearchProps {
     onChange: (searchTerm: string) => void;
+    onAfterChange?: () => void;
 }
 
-export default function MappingSearch({onChange}: MappingSearchProps) {
+export default function MappingSearch({onChange, onAfterChange = () => {}}: MappingSearchProps) {
 
     const [searchString, setSearchString] = useState('');
 
     const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
     const debouncedSearchValue = useDebounce(searchString);
 
+    const memoizedOnChange = React.useCallback(onChange, []);
```

Comment:
issue (blocking):  The mappingSearch doesn't need to memoize the onChange, the components that use MappingSearch most likely will want to.